File: app/src/main/java/humbleactivity/app/Presenter.java
package humbleactivity.app;

import rx.subscriptions.CompositeSubscription;
import timber.log.Timber;

public abstract class Presenter<ViewType extends PassiveView> {
    protected ViewType view;
    protected CompositeSubscription subscriptions;

    public void attach(ViewType view) {
        Timber.d("Attaching %s", view.toString());
        this.view = view;
        this.subscriptions = new CompositeSubscription();
    }

    public void detach() {
        Timber.d("Detaching %s", view.toString());
        subscriptions.unsubscribe();
        view = null;
    }
}

File: app/src/main/java/humbleactivity/app/ui/Presenter.kt
package humbleactivity.app.ui

import rx.subscriptions.CompositeSubscription
import timber.log.Timber

abstract class Presenter<ViewType : PassiveView> {
    protected var view: ViewType? = null
    protected lateinit var subscriptions: CompositeSubscription

    open fun attach(view: ViewType) {
        Timber.d("Attaching %s", view.toString())
        this.view = view
        this.subscriptions = CompositeSubscription()
    }

    fun detach() {
        Timber.d("Detaching %s", view!!.toString())
        subscriptions.unsubscribe()
        view = null
    }
}
